ildcp-core/src/main/java/org/interledger/ildcp/IldcpRequest.java

```diff
@@ -36,11 +38,19 @@ static IldcpRequestBuilder builder() {
   /**
    * An arbitrary expiration date for this packet, as configured by the sender.
    */
-  Instant getExpiresAt();
+  default Instant getExpiresAt() {
+    return Instant.now().plusSeconds(30);
+  }
 
   @Immutable
   abstract class AbstractIldcpRequest implements IldcpRequest {
 
+    @Override
+    @Default
+    public Instant getExpiresAt() {
```

Comment:
Immutable doesn't play nicely with Java 8 default interface methods ?

Reply:
Just maintaining the existing style of the project.

I think the original intent was to define interfaces that had no mention of Immutables, and the define an `abstract` class extension of the interfaces so that future code could create alternative implementations, potentially without needing to depend on the `immutables` library.

That said, as a pattern, it has mostly just created code-duplication. The `immutables` dependency isn't strictly needed at runtime, and having it as a `provided` dependency isn't that big of a deal. I wouldn't be opposed to moving to the "interface-only" style, but it's probably something we should do in another ticket, all at once.

Reply:
Thanks the info